File: original_examples/Codes4Unit2/coin.m
%COIN
close all
p = 0:0.01:1;
P10T = @(p) 11 * (1-p).^10; % 11 to normalize to a density
figure(1)
plot(p, P10T(p), '-', 'LineWidth',2)
% 
quad(P10T, 0, 1)
%
%for the mean
pP10T = @(p)  11 * p .* (1-p).^10;
quad(pP10T, 0, 1) %0.083333 (1/12)
hold on
scatter([0 0],[0 11],200,'r','filled')
scatter([1/12 1/12],[0  P10T(1/12)],200,'g','filled')
% for the median
quad(P10T, 0, 0.061069)  %0.50000
scatter([0.061069 0.061069],[0  P10T(0.061069)],...
        200,'b','filled')
    title ('Coin');
xlabel ('p');
ylabel ('f(p)');
  text (0.05, 11, '\fontsize{14} mode');
legend ('\fontsize{12} Likelihood','\fontsize{12} MAP',...
    '\fontsize{12} Mean','\fontsize{12} Median');    

File: original_examples/Codes4Unit2/coin.py
# -*- coding: utf-8 -*-
"""
Created on Tue Sep  5 22:18:07 2017
@author: bv20
"""

import numpy as np
import matplotlib.pyplot as plt
import scipy.integrate
#COIN

plt.figure()
p = np.arange(0,1,0.001)
p10t = lambda p: 11 * (1-p)**10
plt.plot(p, p10t(p),label='density')
 
area = scipy.integrate.quad(p10t, 0, 1)
print(area)
#
#for the mean
pp10t = lambda p: 11 * p * (1-p)**10 
mean=scipy.integrate.quad(pp10t, 0, 1)
print(mean)
# for the median
is05 = scipy.integrate.quad(p10t, 0, 0.0610690893384)
print(is05)
#
plt.scatter((0,0), (0,11), color = 'r', alpha=0.5, s=100, label='mode')
plt.scatter((1/12,1/12), (0,p10t(1/12)), color = 'g', alpha=0.5, s=100, label='mean')
x=(0.06107, 0.06107); y=(0,  p10t(0.06107))
plt.scatter(x, y, color = 'k', alpha=0.5, s=100, label='median')
plt.legend(loc='upper right')
plt.show()
